### PR TITLE
Added grammar of comments

### PR DESCRIPTION
**Pull request by brauliobz**
Issue #84

### CODE REVIEW

File: src/comments.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Comments
 
+> **<sup>Lexer</sup>**  
+> LINE_COMMENT :  
+> &nbsp;&nbsp; `//` ~[\n\r]*
```
Comment (qnighy) [inline src/comments.md]: A line comment can include bare CRs unless it isn't a doc-comment.

File: src/comments.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
 
+> **<sup>Lexer</sup>**  
+> LINE_COMMENT :  
+> &nbsp;&nbsp; `//` ~[\n\r]*  
+>  
+> BLOCK_COMMENT :  
+> &nbsp;&nbsp; `/*` (BLOCK_COMMENT | .)* `*/`  
+>  
+> OUTER_DOC_LINE_COMMENT :  
+> &nbsp;&nbsp; `//!` ~[\n\r]*
```
Comment (qnighy) [inline src/comments.md]: Only one of LF/CRLF/EOF can end outer/inner doc line comments and bare CR isn't allowed here. For example, `"//! foo \r//! bar\n"` will lead to a syntax error. The difficulty here would be to express the fact that `//! foo \r//! bar\n` isn't a non-doc comment neither.

File: src/comments.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+> &nbsp;&nbsp; `/*` (BLOCK_COMMENT | .)* `*/`  
+>  
+> OUTER_DOC_LINE_COMMENT :  
+> &nbsp;&nbsp; `//!` ~[\n\r]*  
+>  
+> OUTER_DOC_BLOCK_COMMENT :  
+> &nbsp;&nbsp; `/*!` (OUTER_DOC_BLOCK_COMMENT | .)* `*/`  
+>  
+> INNER_DOC_LINE_COMMENT :  
+> &nbsp;&nbsp; `///` ~[\n\r]*
```
Comment (qnighy) [inline src/comments.md]: There is an exception: comments starting with `////` is a non-doc line comment.

File: src/comments.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+> &nbsp;&nbsp; `//` ~[\n\r]*  
+>  
+> BLOCK_COMMENT :  
+> &nbsp;&nbsp; `/*` (BLOCK_COMMENT | .)* `*/`  
+>  
+> OUTER_DOC_LINE_COMMENT :  
+> &nbsp;&nbsp; `//!` ~[\n\r]*  
+>  
+> OUTER_DOC_BLOCK_COMMENT :  
+> &nbsp;&nbsp; `/*!` (OUTER_DOC_BLOCK_COMMENT | .)* `*/`
```
Comment (qnighy) [inline src/comments.md]: In outer/inner doc block comments, any kinds of block comments `/* */` `/** */` `/*! */` can be nested.

Comment (qnighy) [inline src/comments.md]: In outer/inner doc comments, bare CR isn't allowed. CRLF is allowed instead.

**Comment by brauliobz** [comment]
@qnighy Thank you for your comments :+1: 

I changed the comments' grammar to account for the special cases (and put a lot of examples to make these clear).

I'll try to tackle the CR / CRLF problem tomorrow.

**Comment by Havvy** [comment]
@brauliobz What's the status here?

**Comment by brauliobz** [comment]
@Havvy I got involved with other parts of the grammar and forgot about this one :/ . The remaining problem here is the CR/LF details. I'm doing it now.

**Comment by brauliobz** [comment]
[Rendered here.](https://brauliobz.github.io/rust-reference/comments.html)

**Comment by steveklabnik** [comment]
LGTM, thanks!

**Pull request merged**